v3-sdk/quoting/src/example/Example.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useCallback } from 'react'
+import './Example.css'
+import { ethers } from 'ethers'
+import { Token } from '@uniswap/sdk-core'
+import { CurrentConfig } from '../config'
+import { computePoolAddress } from '@uniswap/v3-sdk'
+import Quoter from '@uniswap/v3-periphery/artifacts/contracts/lens/Quoter.sol/Quoter.json'
+import IUniswapV3PoolABI from '@uniswap/v3-core/artifacts/contracts/interfaces/IUniswapV3Pool.sol/IUniswapV3Pool.json'
+import { POOL_FACTORY_CONTRACT_ADDRESS, QUOTER_CONTRACT_ADDRESS } from '../libs/constants'
+import { getProvider } from '../libs/providers'
+import { toReadableAmount, fromReadableAmount } from '../libs/conversion'
+
+const getPoolConstants = async (): Promise<{ token0: string; token1: string; fee: number }> => {
+  const provider = getProvider()
+  if (!provider) {
+    throw new Error('Provider not found')
+  }
+  const currentPoolAddress = computePoolAddress({
+    factoryAddress: POOL_FACTORY_CONTRACT_ADDRESS,
+    tokenA: CurrentConfig.tokens.in as Token,
+    tokenB: CurrentConfig.tokens.out as Token,
+    fee: CurrentConfig.tokens.fee,
+  })
+
+  const poolContract = new ethers.Contract(currentPoolAddress, IUniswapV3PoolABI.abi, provider)
+  const [token0, token1, fee] = await Promise.all([poolContract.token0(), poolContract.token1(), poolContract.fee()])
+
+  return {
+    token0,
+    token1,
+    fee,
+  }
+}
+
+const quote = async (): Promise<number> => {
+  const provider = getProvider()
+  if (!provider) {
+    throw new Error('Provider not found')
+  }
+  const quoterContract = new ethers.Contract(QUOTER_CONTRACT_ADDRESS, Quoter.abi, provider)
+  const poolConstants = await getPoolConstants()
+
+  const quotedAmountOut = await quoterContract.callStatic.quoteExactInputSingle(
+    poolConstants.token0,
+    poolConstants.token1,
+    poolConstants.fee,
+    fromReadableAmount(CurrentConfig.tokens.amountIn, CurrentConfig.tokens.in.decimals),
+    0
+  )
+
+  return toReadableAmount(quotedAmountOut, CurrentConfig.tokens.out.decimals)
+}
+
+const Example = () => {
+  const [outputAmount, setOutputAmount] = useState<number>()
+
+  const onTrade = useCallback(async () => {
+    setOutputAmount(await quote())
+  }, [])
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        {CurrentConfig.rpc.mainnet === '' && <h2 className="error">Please set your mainnet RPC URL in config.ts</h2>}
+        <h3>{`Quote input amount: ${CurrentConfig.tokens.amountIn} ${CurrentConfig.tokens.in.symbol}`}</h3>
+        <h3>{`Quote output amount: ${outputAmount} ${CurrentConfig.tokens.out.symbol}`}</h3>
+        <button onClick={onTrade}>
+          <p>Trade</p>
```

Comment:
This got switched back to Trade vs Quote at some point, should fix button and handler name